test/tools/libtesteth/BlockChainHelper.cpp

```diff
@@ -273,6 +273,14 @@ void TestBlock::mine(TestBlockChain const& _bc)
     }
     else
         recalcBlockHeaderBytes();
+
+    // Create dummy signature for NoProof blocks
+    if (Ethash::mixHash(m_blockHeader) == h256{0})
+    {
+        m_blockHeader.setSeal(Ethash::NonceField, h64{0});
+        m_blockHeader.setSeal(Ethash::MixHashField, h256{0});
```

Comment:
Why do you assign it here if you just checked that it's 0?

Reply:
because it returns 0 also when there is no signature set.

Reply:
We could implenent a method hasSignature() if you woukd like. That will change libethereum. By default I try not to touch libethereum for test related logic